pytorch_lightning/profiler/profilers.py

```diff
@@ -55,6 +41,38 @@ def start(self, action_name: str) -> None:
     def stop(self, action_name: str) -> None:
         """Defines how to record the duration once an action is complete."""
 
+    @abstractmethod
+    def summary(self) -> str:
+        """Create profiler summary in text format."""
+
+
+class BaseProfiler(AbstractProfiler, ABC):
+    """
+    If you wish to write a custom profiler, you should inherit from this class.
+    """
+
+    def __init__(self, local_rank: Optional[int] = None, log_dir: Optional[str] = None) -> None:
+        self.output_fname = getattr(self, "output_fname", None)
+        self.output_file = None
+        # the profiler can be used outside of lightning
+        # that's why we call `on_train_start` manually
+        self.on_train_start(local_rank=local_rank, log_dir=log_dir)
```

Comment:
i think it'd be natural to also ask users doing this outside of lightning to call the same hooks rather than calling this in the `init` similar to how the datamodule offers `prepare_data/setup`

Reply:
Yes, it sounds wrong as profiling should also work for test and predict. Let me think on how to clean this.

Reply:
Yeah, let's make it implement setup and teardown instead

Reply:
Marking as resolved: continue discussion in https://github.com/PyTorchLightning/pytorch-lightning/pull/6621